src/Stratis.Bitcoin.IntegrationTests/EnvironmentMockUpHelpers/NodeRunners.cs

```diff
@@ -192,33 +262,70 @@ public static FullNode BuildFullNode(NodeSettings args, Action<IFullNodeBuilder>
 
             return node;
         }
-
-        public FullNode FullNode;
     }
 
-    public static class FullNodeTestBuilderExtension
+    public sealed class StratisProofOfWorkMiningNode : INodeRunner
     {
-        public static IFullNodeBuilder MockIBD(this IFullNodeBuilder fullNodeBuilder)
+        private Action<IFullNodeBuilder> callback;
+        public FullNode FullNode { get; set; }
+        private bool mineCoinsFast;
+
+        public StratisProofOfWorkMiningNode(Action<IFullNodeBuilder> callback = null) : base()
+        {
+            this.callback = callback;
+        }
+
+        public bool IsDisposed
+        {
+            get { return this.FullNode.State == FullNodeState.Disposed; }
+        }
+
+        public void Kill()
+        {
+            this.FullNode?.Dispose();
+        }
+
+        public void Start(string dataDir, bool mineCoinsFast)
         {
-            fullNodeBuilder.ConfigureFeature(features =>
+            NodeSettings nodeSettings = new NodeSettings(Network.StratisRegTest, ProtocolVersion.ALT_PROTOCOL_VERSION, args: new string[] { "-conf=stratis.conf", "-datadir=" + dataDir }, loadConfiguration: false);
+
+            this.mineCoinsFast = mineCoinsFast;
+
+            var node = BuildFullNode(nodeSettings, this.callback);
+
+            this.FullNode = node;
+            this.FullNode.Start();
+        }
+
+        public FullNode BuildFullNode(NodeSettings args, Action<IFullNodeBuilder> callback = null)
+        {
+            FullNode node;
+
+            if (callback != null)
+            {
+                var builder = new FullNodeBuilder().UseNodeSettings(args);
+                callback(builder);
+                node = (FullNode)builder.Build();
+            }
+            else
             {
-                foreach (IFeatureRegistration feature in features.FeatureRegistrations)
-                {
-                    feature.FeatureServices(services =>
-                    {
-                        // Get default IBD implementation and replace it with the mock.
-                        ServiceDescriptor ibdService = services.FirstOrDefault(x => x.ServiceType == typeof(IInitialBlockDownloadState));
-
-                        if (ibdService != null)
-                        {
-                            services.Remove(ibdService);
-                            services.AddSingleton<IInitialBlockDownloadState, InitialBlockDownloadStateMock>();
-                        }
-                    });
-                }
-            });
-
-            return fullNodeBuilder;
+                var builder = new FullNodeBuilder()
+                    .UseNodeSettings(args)
+                    .UsePosConsensus()
+                    .UseBlockStore()
+                    .UseMempool()
+                    .AddMining()
```

Comment:
use here `AddPosPowMining`

Reply:
We can only use `AddMining` as `AddPosPowMining` uses the PosAssemblerFactory which uses `PosMinting`.

Reply:
`PosAssemblerFactory` does not necessarily use `PosMinting`, but if you use it to make a PoW block the coinbase transaction is hardcoded to be empty. This is a bug (1113#).